packages/caver-wallet/src/keyringContainer.js

```diff
@@ -0,0 +1,301 @@
+     */
+    add(keyring) {
+        if (this._addressKeyringMap.get(keyring.address.toLowerCase()) !== undefined)
+            throw new Error(`Duplicate Account ${keyring.address}. Please use updateKeyring() instead.`)
+
+        const keyringToAdd = keyring.copy()
+
+        this._addressKeyringMap.set(keyringToAdd.address.toLowerCase(), keyringToAdd)
+
+        return keyringToAdd
+    }
+
+    /**
+     * deletes the keyring that associates with the given address from keyringContainer.
+     *
+     * @param {string} address An address of the keyring to be deleted in keyringContainer.
+     * @return {boolean}
+     */
+    remove(address) {
+        let keyringToRemove
+        if (utils.isAddress(address)) {
+            keyringToRemove = this.getKeyring(address)
+        } else {
+            throw new Error(`To remove the keyring, the first parameter should be an address string.`)
+        }
+
+        if (keyringToRemove === undefined) return false
+
+        // deallocate keyring object created for keyringContainer
+        keyringToRemove.key = null
+        this._addressKeyringMap.delete(keyringToRemove.address.toLowerCase())
+
+        return true
+    }
+
+    /**
+     * signs with data and returns the result object that includes `signature`, `message` and `messageHash`
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {string} data The data string to sign.
+     * @param {number} [role] A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] An index of key to use for signing.
+     * @return {object}
+     */
+    signMessage(address, data, role, index) {
+        const keyring = this.getKeyring(address)
+        if (keyring === undefined) throw new Error(`Failed to find keyring from wallet with ${address}`)
+        return keyring.signMessage(data, role, index)
+    }
+
+    /**
+     * signs the transaction using one key and return the transactionHash
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {Transaction} transaction A transaction object.
+     * @param {number} [index] An index of key to use for signing.
+     * @param {function} [hasher] A function to return hash of transaction. In order to use a custom hasher, the index must be defined.
+     * @return {Transaction}
+     */
+    async signWithKey(address, transaction, index = 0, hasher = TransactionHasher.getHashForSigning) {
+        if (!transaction.from || transaction.from === '0x') transaction.from = address
+        if (transaction.from.toLowerCase() !== address.toLowerCase())
+            throw new Error(
+                `transaction.from ${transaction.from.toLowerCase()} is different from the given address ${address.toLowerCase()}.`
+            )
+
+        // Optional parameter cases
+        // (address transaction) / (address transaction index) / (address transaction index hasher)
```

Comment:
I think the comment should be changed as well.

Reply:
I modified parameter cases. Is there anything wrong???

Reply:
Actually, we now don't have any optional parameters.
Hmm.. It is not very clear, but we now always have four parameters with default parameters index and hasher. We don't have any case that we have only `(address transaction)`.

Reply:
I think, parameter which has default value also optional parameter to users.
I understood what you mean, so i think change this comment to "User parameter input cases" will be fine.